pkg/sql/pgwire/pgerror/errors.go

```diff
@@ -132,6 +133,39 @@ func GetPGCause(err error) (*Error, bool) {
 	}
 }
 
+// NewAssertionErrorf creates an internal error.
+func NewAssertionErrorf(format string, args ...interface{}) error {
+	err := NewErrorWithDepthf(1, CodeInternalError, "programming error: "+format, args...)
```

Comment:
I'd say "internal error" here or -- "programming" isn't very specific on its own, e.g. most of the things interacting with a DB are also "programming", so it could be interpreted as a usage error.